hercules/plant_components/wind_farm.py

```diff
@@ -1,43 +1,80 @@
-
+# Unified wind farm model for Hercules supporting multiple wake modeling strategies.
 
 import numpy as np
 import pandas as pd
-from floris import FlorisModel
+from floris import ApproxFlorisModel, FlorisModel
+from floris.core import average_velocity
+from floris.uncertain_floris_model import map_turbine_powers_uncertain
 from hercules.plant_components.component_base import ComponentBase
 from hercules.utilities import (
     hercules_float_type,
     interpolate_df,
     load_perffile,
     load_yaml,
 )
-from scipy.interpolate import interp1d
 from scipy.optimize import minimize_scalar
 from scipy.stats import circmean
 
 RPM2RADperSec = 2 * np.pi / 60.0
 
 
-class Wind_MesoToPower(ComponentBase):
-    def __init__(self, h_dict):
-        """Initialize the Wind_MesoToPower class.
+class WindFarm(ComponentBase):
+    """Unified wind farm model with configurable wake modeling strategies.
+
+    This model simulates wind farm performance by applying wind speed time signals
+    to turbine models. It supports three wake modeling strategies:
+
+    1. **dynamic**: Real-time FLORIS wake calculations at each time step or interval.
+       Use when turbines may have individual setpoints or non-uniform operation.
 
-        This model focuses on meso-scale wind phenomena by applying a separate wind speed
-        time signal to each turbine model derived from data. It combines FLORIS wake
-        modeling with detailed turbine dynamics for wind farm performance analysis.
+    2. **precomputed**: Pre-computed FLORIS wake deficits for all conditions.
+       Use when all turbines operate uniformly (all on, all off, or uniform curtailment).
+       More efficient but less flexible than dynamic.
+
+    3. **none**: No wake modeling - wind speeds are used directly.
+       Use when wake effects are already included in the input data or when
+       wake modeling is not needed.
+
+    All three strategies support detailed turbine dynamics (filter_model or dof1_model).
+    """
+
+    def __init__(self, h_dict, wake_model=None):
+        """Initialize the WindFarm class.
 
         Args:
-            h_dict (dict): Dictionary containing values for the simulation.
+            h_dict (dict): Dictionary containing simulation parameters.
+            wake_model (str, optional): Wake modeling strategy: "dynamic", "precomputed",
+                or "no_added_wakes". If None, infers from component_type for backward compatibility.
+                Defaults to None.
+
+        Raises:
+            ValueError: If wake_model is invalid or required parameters are missing.
         """
         # Store the name of this component
         self.component_name = "wind_farm"
 
-        # Store the type of this component
-        self.component_type = "Wind_MesoToPower"
+        # Infer wake_model from component_type if not provided (backward compatibility)
+        if wake_model is None:
+            wake_model = self._infer_wake_model_from_component_type(h_dict)
```

Comment:
I don't think this (or the `self._infer_wake_model_from_component_type` method) is needed? The only place `WindFarm` is instantiated is in `HybridPlant`, where we always pass `wake_model` directly.

Reply:
Removed method